sdks/java/testing/nexmark/src/main/java/org/apache/beam/sdk/nexmark/NexmarkUtils.java

```diff
@@ -145,53 +145,51 @@
     QUERY_RUNNER_AND_MODE
   }
 
+  /** Return a query name with query language (if applicable). */
+  static String fullQueryName(String queryLanguage, String query) {
+    return queryLanguage != null ? query + "_" + queryLanguage : query;
+  }
+
   /** Return a BigQuery table spec. */
   static String tableSpec(NexmarkOptions options, String queryName, long now, String version) {
+    return String.format(
+        "%s:%s.%s",
+        options.getProject(),
+        options.getBigQueryDataset(),
+        tableName(options, queryName, now, version));
+  }
+
+  /** Return a BigQuery table name. */
+  static String tableName(NexmarkOptions options, String queryName, long now, String version) {
     String baseTableName = options.getBigQueryTable();
     if (Strings.isNullOrEmpty(baseTableName)) {
       throw new RuntimeException("Missing --bigQueryTable");
     }
+
     switch (options.getResourceNameMode()) {
       case VERBATIM:
-        return String.format(
-            "%s:%s.%s_%s",
-            options.getProject(), options.getBigQueryDataset(), baseTableName, version);
+        return String.format("%s_%s", baseTableName, version);
       case QUERY:
-        return String.format(
-            "%s:%s.%s_%s_%s",
-            options.getProject(), options.getBigQueryDataset(), baseTableName, queryName, version);
+        return String.format("%s_%s_%s", baseTableName, queryName, version);
       case QUERY_AND_SALT:
-        return String.format(
-            "%s:%s.%s_%s_%s_%d",
-            options.getProject(),
-            options.getBigQueryDataset(),
-            baseTableName,
-            queryName,
-            version,
-            now);
+        return String.format("%s_%s_%s_%d", baseTableName, queryName, version, now);
       case QUERY_RUNNER_AND_MODE:
-        return (version != null)
```

Comment:
I only refactored this so I wanted to keep the original functionality but the way it was (and is) implemented is that we check version nullability only in `QUERY_AND_RUNNER_MODE`. It seems that it should be checked in every branch of this switch, otherwise we can have `null` versions in table names (eg. for "verbatim"). 

Maybe some of the switch options do not require version at all?
Should I fix this too in this PR?

Reply:
QUERY_AND_RUNNER_MODE was added for nexmark publish perfs to BQ and is always called with null version by nexmark. It is the only code patch where version is null, all the other modes are always called with non null version. I added the check just in case a user sets `ResourceNameMode` to QUERY_AND_RUNNER_MODE but uses tableSpec for another output (to kafka, pubsub ...) and in that case the version will be not null. 

But I'm fine with adding the null check for all the modes even if with current code, version cannot be null in some modes.